server/fleet/app.go

```diff
@@ -270,29 +148,216 @@ type SMTPSettingsPayload struct {
+		return defaultOrZeroFunc()
+	}
+
+	switch typed := field.(type) {
+	case int:
+		return typed
+	case *int:
+		if typed == nil {
+			return defaultOrZeroFunc()
+		}
+		return *typed
+	default:
+		return defaultOrZeroFunc()
+	}
+}
+
+func (c *AppConfig) GetUint(path string) uint {
+	field := c.Get(path)
+	defaultOrZeroFunc := func() uint {
+		defaultFunc, ok := configDefaults[path]
+		if !ok {
+			return 0
+		}
+		return defaultFunc().(uint)
+	}
+	if !reflect.ValueOf(field).IsValid() {
+		return defaultOrZeroFunc()
+	}
+
+	switch typed := field.(type) {
+	case uint:
+		return typed
+	case *uint:
+		if typed == nil {
+			return defaultOrZeroFunc()
+		}
+		return *typed
+	default:
+		return defaultOrZeroFunc()
+	}
+}
+
+func (c *AppConfig) GetBool(path string) bool {
+	field := c.Get(path)
+	defaultOrZeroFunc := func() bool {
+		defaultFunc, ok := configDefaults[path]
+		if !ok {
+			return false
+		}
+		return defaultFunc().(bool)
+	}
+	if !reflect.ValueOf(field).IsValid() {
+		return defaultOrZeroFunc()
+	}
+
+	switch typed := field.(type) {
+	case bool:
+		return typed
+	case *bool:
+		if typed == nil {
+			return defaultOrZeroFunc()
+		}
+		return *typed
+	default:
+		return defaultOrZeroFunc()
+	}
+}
+
+func (c *AppConfig) GetString(path string) string {
+	field := c.Get(path)
+	defaultOrZeroFunc := func() string {
+		defaultFunc, ok := configDefaults[path]
+		if !ok {
+			return ""
+		}
+		return defaultFunc().(string)
+	}
+	if !reflect.ValueOf(field).IsValid() {
+		return defaultOrZeroFunc()
+	}
+
+	switch typed := field.(type) {
+	case string:
+		return typed
+	case *string:
+		if typed == nil {
+			return defaultOrZeroFunc()
+		}
+		return *typed
+	default:
+		return defaultOrZeroFunc()
+	}
+}
+
+func (c *AppConfig) GetJSON(path string) json.RawMessage {
+	field := c.Get(path)
+	defaultOrZeroFunc := func() json.RawMessage {
+		defaultFunc, ok := configDefaults[path]
+		if !ok {
+			return nil
+		}
+		return defaultFunc().(json.RawMessage)
+	}
+	if !reflect.ValueOf(field).IsValid() {
+		return defaultOrZeroFunc()
+	}
+
+	switch typed := field.(type) {
+	case json.RawMessage:
+		return typed
+	case *json.RawMessage:
+		if typed == nil {
+			return defaultOrZeroFunc()
+		}
+		return *typed
+	default:
+		return defaultOrZeroFunc()
+	}
+}
+
+func (c *AppConfig) GetStringPtr(path string) *string {
+	return ptr.String(c.GetString(path))
+}
+
+func (c *AppConfig) GetBoolPtr(path string) *bool {
+	return ptr.Bool(c.GetBool(path))
+}
+
+func (c *AppConfig) GetUintPtr(path string) *uint {
+	return ptr.Uint(c.GetUint(path))
+}
+
+func (c *AppConfig) GetIntPtr(path string) *int {
+	return ptr.Int(c.GetInt(path))
+}
+
+func (c *AppConfig) GetJSONPtr(path string) *json.RawMessage {
+	return ptr.RawMessage(c.GetJSON(path))
+}
+
+func getFieldByPath(obj interface{}, path ...string) interface{} {
+	// We ignore the errors here because all the errors in reflections are based on type, we know what types are
+	// going through here.
```

Comment:
I looked at how viper handles errors, and basically they ignore them. So I figured, since we are working with a data structure we know about, we can also afford to ignore them and have a better API.